src/change/models/merger.ts

```diff
@@ -64,7 +64,7 @@ const mergeChanges = (changes: ChangeWithMetadata[], changesetId: number): [OsmX
     version: '0.6',
     create: { node: [], way: [] },
     modify: { node: [], way: [] },
-    delete: { node: [], way: [] },
+    delete: { way: [], node: [] },
```

Comment:
our scenario is very obscured, therefore we better add a comment why the order matters